modules/infra/datapath/control_output.c

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2024 Christophe Fontaine
+
+#include "gr_control_output.h"
+
+#include <gr_control.h>
+#include <gr_graph.h>
+#include <gr_log.h>
+#include <gr_mbuf.h>
+
+#include <rte_ether.h>
+#include <rte_graph_worker.h>
+
+#include <stdlib.h>
+
+enum {
+	CONTROL_OUTPUT_ERROR,
+	EDGE_COUNT,
+};
+
+struct rte_ring *control_output_ring;
```

Comment:
static?

Reply:
This variable should be removed from here. See my previous review.

Reply:
fixed in the newer version, `control_output_push` is now hiding the ring.